parquet/metadata/file.go

```diff
@@ -243,11 +243,13 @@ type FileMetaData struct {
 	// size of the raw bytes of the metadata in the file which were
 	// decoded by thrift, Size() getter returns the value.
 	metadataLen int
+
+	footerOffset int64
 }
 
 // NewFileMetaData takes in the raw bytes of the serialized metadata to deserialize
 // and will attempt to decrypt the footer if a decryptor is provided.
-func NewFileMetaData(data []byte, fileDecryptor encryption.FileDecryptor) (*FileMetaData, error) {
+func NewFileMetaData(data []byte, footerOffset int64, fileDecryptor encryption.FileDecryptor) (*FileMetaData, error) {
```

Comment:
The only thing I'm not a fan of here is that this is a breaking change. Now, I don't think there are many (if any) packages that are importing and using this function directly (as creating new file metadata would primarily be done via the Writer rather than done directly) but it's still a concern that I'd rather not create a breaking change if we can avoid it. 

Is there a way we can avoid this being a breaking change at all? If 0 is valid, maybe we create a setter?

Reply:
Yes, I forgot the backwards compatibility. Can we name the setter/getter function like:

```go
// SetSourceFileSize get the total size of the source file from meta data.
func (f *FileMetaData) SetSourceFileSize() int64 { return f.sourceFileSize }

// GetSourceFileSize set the total size of the source file in meta data.
func (f *FileMetaData) GetSourceFileSize() int64 { return f.sourceFileSize }

// file.go
type FileMetaData struct {
    ....
    
    // sourceFileSize is not a part of FileMetaData, but it is mainly used to parse meta data.
    // Users can manually set this value and they are responsible for the validity of it.
    sourceFileSize int64
}
```